### DISCUSSION TITLE
connect mapProps 中的第二个参数类型是 GeneralField会ts 报错

### DISCUSSION THREAD

**Discussion by happyElina** [alibaba/formily#1702] [Q&A 2.x] [answered]
`const FormItem = connect(
  Form.Item,
  mapProps(
    {
      title: 'label',
      description: 'extra',
      required: true,
      validateStatus: true,
    },
    (props, field) => { // 这个field的类型不对，导致field.errors报错
      return {
        ...props,
        help: field.errors?.length ? field.errors : undefined,
      }
    }
  )
)`

ps : 我跑的就是https://react.formilyjs.org/api/shared/connect 里的demo代码

**Comment by janryWang** [comment, marked as answer]
用isField/isVoidField这种判断，需要做类型守护的
